### PR TITLE
fix(networkport): correctly display metrics

### PR DESCRIPTION
**Pull request by stonebuzz**
Correctly display NetworkPort metrics

image

- Fix load of chartist js library 
- Convert bytes to megabytes (dashboard label more readable)

| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #12472

### CODE REVIEW

**Comment by stonebuzz** [comment]
Same error as #11878
From chrome web browser
 
image

File: src/NetworkPortMetrics.php [hunk 1]
```diff
@@ -142,27 +142,24 @@ public function showMetrics(NetworkPort $netport)
...
             foreach ($errors_metrics as $key => $value) {
                 $errors_series[$key]['name'] = $this->getLabelFor($key);
-                $errors_series[$key]['data'][] = $value;
+                $errors_series[$key]['data'][] = round($value / 1024 / 1024, 0); //convert bytes to megabytes
             }
-
-            ++$i;
         }
 
         $bytes_bar_conf = [
```
Comment (cconard96) [inline src/NetworkPortMetrics.php:165]: The label in the `bytes_bar_conf` would need updated to indicate that the data is in megabytes now.

**Review by cconard96 — DISMISSED** [review]

**Comment by cconard96** [comment]
> Same error as #11878 From chrome web browser
> 
> image

For me, it had trouble parsing the chart height. It expected a number or a string starting with a number and ignoring everything else (probably to support something like '100px'?), but was getting a CSS `calc` statement. It didn't make any sense to me though since I hadn't seen any issues with widgets on the dashboards.

**Comment by stonebuzz** [comment]
maybe -> https://github.com/chartist-js/chartist/issues/1099

File: src/NetworkPortMetrics.php (hunk 1 above)
Comment (stonebuzz) [reply src/NetworkPortMetrics.php:165]: fixed

**Comment by cconard96** [comment]
Commenting out the draw listener that contains the animate call didn't resolve the issue. Only removing the height option or specifying a numeric value worked.
https://github.com/glpi-project/glpi/blob/ae93c956786a2021b0b27b300ee999f993167d47/src/Dashboard/Widget.php#L1588

**Comment by stonebuzz** [comment]
> Commenting out the draw listener that contains the animate call didn't resolve the issue. Only removing the height option or specifying a numeric value worked.
> 
> https://github.com/glpi-project/glpi/blob/ae93c956786a2021b0b27b300ee999f993167d47/src/Dashboard/Widget.php#L1588

I confirm

**Comment by cconard96** [comment]
I think I found a solution that would remove the need for a calculated height and seems to allow both the network port metrics and dashboard line charts to display correctly.

1. Add a container div around the `.chart` one so that the automatic height is 100% of this element and not the one that also contains the `.main-label` and `.main-icon` elements (when within a dashboard).
2. Remove the height option when initializing the chart
3. I think the style that uses the calculated height could be removed too. I didn't notice a difference in Chrome after making the other changes.

```diff
diff --git a/src/Dashboard/Widget.php b/src/Dashboard/Widget.php
index d78e8a6e95..1a1aa92d6f 100644
--- a/src/Dashboard/Widget.php
+++ b/src/Dashboard/Widget.php
@@ -1513,20 +1513,14 @@ JAVASCRIPT;
             })";
         }
 
-        $height = "calc(100% - 1px)";
         $legend_options = "";
         if ($p['legend']) {
-            $height = "calc(100% - 40px)";
             $legend_options = "
             Chartist.plugins.legend(),";
         }
 
         $html = <<<HTML
       <style>
-         /** fix chrome resizing height when animating svg (don't know why) **/
-      #{$chart_id} .ct-chart-line {
-         min-height: $height;
-      }
 
       #{$chart_id} {
          background-color: {$p['color']};
@@ -1564,11 +1558,13 @@ JAVASCRIPT;
       {$palette_style}
       </style>
 
-      <div class="card g-chart $class"
-           id="{$chart_id}">
-         <div class="chart ct-chart"></div>
-         <span class="main-label">{$p['label']}</span>
-         <i class="main-icon {$p['icon']}"></i>
+      <div>
+          <div class="card g-chart $class"
+               id="{$chart_id}">
+             <div class="chart ct-chart"></div>
+             <span class="main-label">{$p['label']}</span>
+             <i class="main-icon {$p['icon']}"></i>
+          </div>
       </div>
 HTML;
 
@@ -1585,7 +1581,6 @@ HTML;
             series: {$json_series},
          }, {
             width: '100%',
-            height: '{$height}',
             fullWidth: true,
             chartPadding: {
                right: 40

```

**Review by trasher — APPROVED** [review]

**Comment by orthagh** [comment]
> I think I found a solution that would remove the need for a calculated height and seems to allow both the network port metrics and dashboard line charts to display correctly.

I feared that it could broke the advanceddashboard plugin where we had some issue with height but it seems everything work without flaws, so lgtm.

**Comment by orthagh** [comment]
To note, in `Widget.php` there is other methods relying on this calculated height and should be fixed in the same way

**Comment by stonebuzz** [comment]
> To note, in `Widget.php` there is other methods relying on this calculated height and should be fixed in the same way

I've listed :
- getBarsGraph
- pie

**Review by orthagh — APPROVED** [review]

**Review by cconard96 — APPROVED** [review]

**Pull request merged**